From 5f859b30d34c3e1ec3fca48d55248b502669fcc5 Mon Sep 17 00:00:00 2001
From: Craig Small <csmall@dropbear.xyz>
Date: Tue, 22 Dec 2020 16:08:49 +1100
Subject: [PATCH] pgrep: Remove memory leak

This is part of !118 where @tt.rantala found a memory leak.
The other part of !118 may come later if the performance change
is significant.

References:
 procps-ng/procps!118
---
 pgrep.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/pgrep.c b/pgrep.c
index 205a1db..4fe5e8a 100644
--- a/pgrep.c
+++ b/pgrep.c
@@ -716,6 +716,11 @@ static struct el * select_procs (int *num)
         free(cmdsearch);
         free(cmdoutput);
 
+	if (preg) {
+		regfree(preg);
+		free(preg);
+	}
+
 	return list;
 }
 
-- 
2.27.0

